code/modules/autowiki/pages/stockparts.dm

```diff
@@ -0,0 +1,127 @@
+// Autowiki Stock Parts
+/datum/autowiki/stockparts
+	page = "Template:Autowiki/Content/StockParts"
+
+/datum/autowiki/stockparts/generate()
+	// Output string to append to autowiki data block
+	var/output = ""
+
+	// Find all valid stock parts and their references
+	for(var/part_type in subtypesof(/obj/item/stock_parts))
+		// Skip subspace subtype as it is a grouping subtype and invalid
+		if(part_type == /obj/item/stock_parts/subspace)
```

Comment:
This should be done through some `abstract_type` system instead

Reply:
I'm looking into the typecache system so that way I can finally finish this project up, but I'm not sure what you mean by `abstract_type`

Reply:
We have a trick where we specify something like this on types:

```dm
/datum/category
    var/abstract_type = /datum/category

/datum/category/item
    // use as normal
```

...and when iterating, we skip over everything that has `abstract_type == type`